lib/reporting/reports/enterprise_fee_summary/enterprise_fees_with_tax_report_by_order.rb

```diff
@@ -299,6 +305,12 @@ def enterprise_fee_id(query_result_row)
         def keys(query_result_row)
           query_result_row.first
         end
+
+        def apply_voucher_on_amount(order, amount)
+          rate = order.applied_voucher_rate
+          result = amount + (amount * rate)
+          BigDecimal(result.to_s).round(2, BigDecimal::ROUND_HALF_UP)
```

Comment:
It's a shame that we have to round. Unfortunately, there's no better way with the current data structure of vouchers. But I'm pretty sure that we will have rounding errors one day.

Reply:
Yeah I was also thinking that we might face the data round off issues. I was wondering how we can avoid this thing. :sweat_smile:

Reply:
The only solution I can think of is a complete re-modelling of the voucher adjustments. Instead of adjusting the whole order, we could adjust each line item. The total of all voucher adjustments could still be different to the rate applied to the whole order but that's easier to explain. It could also be easier to use those adjustments in reports like this one. But it's a big chunk of work that I wouldn't start right now. Gaetan and Lynne also expressed their concerns over starting this work while there's not big reason for it.